src/s3/builders/bucket_common.rs

```diff
@@ -0,0 +1,64 @@
+// MinIO Rust Library for Amazon S3 Compatible Cloud Storage
+// Copyright 2025 MinIO, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::marker::PhantomData;
+
+use crate::s3::{
+    client::Client,
+    error::Error,
+    utils::{check_bucket_name, Multimap},
+};
+
+#[derive(Clone, Debug, Default)]
+pub struct BucketCommon<A> {
+    pub(crate) client: Option<Client>,
+
+    pub(crate) extra_headers: Option<Multimap>,
+    pub(crate) extra_query_params: Option<Multimap>,
+    pub(crate) region: Option<String>,
+    pub(crate) bucket: String,
+
+    _operation: PhantomData<A>,
+}
+
+impl<A: Default> BucketCommon<A> {
+    pub fn new(bucket: &str) -> Result<BucketCommon<A>, Error> {
+        check_bucket_name(bucket, true)?;
```

Comment:
We can move this check into the individual `to_s3request` trait methods. This lets us avoid returning an error here and leads to an API that is smoother to use in the sense that there is only one place where all errors are returned (in the send() call):

```rust
let r = client.get_bucket_encryption("mybucket").send().await;
// Now handle error value of `r`
```

instead of 
```rust
let r1 = client.get_bucket_encryption("mybuclet");
let r2 = r1.unwrap() // or some actual error handling
let r3 = r2.send().await.unwrap() // or more actual error handling
```